src/docs/sphinx/quickstart.rst

```diff
@@ -267,28 +267,268 @@ must be specified using either:
 
-  $ brew install gcc@8
-  $ brew install llvm@11
-  $ brew install mpich
+   $ sudo port select clang mp-clang-12
+   $ sudo port select mpi openmpi-clang-fortran
+   $ sudo port select gcc mp-gcc12
 
-Build third-party libraries via Uberenv:
+This step tells MacPorts to make symbolic links in your path so that, for example, ``clang`` will invoke the clang12 from the MacPorts package and not the ``clang`` shipped by Apple. 
+It also sets up a set of symlinks so that Clang, GCC, and the MPI wrappers all work without you having to muck with environment variables to locate header files and libraries. 
+It may be possible to skip this step and give full paths to your compilers in ``compilers.yaml`` (instead of the symlinks ``/opt/local/bin/clang``, etc.), but we haven't tried this.
 
-``python3 scripts/uberenv/uberenv.py --spec=%apple-clang@11.0.1 --prefix=../path/to/install``
+While building ParMetis, spack invokes the MPI compiler wrapper with the ``mpic++`` command, but MacPorts does not create this particular synonym for OpenMPI. 
+It does create ``mpicxx``. This can be solved by making a symlink:
 
-.. note::
-   You may need to alter the compiler spec inside ``scripts/spack/configs/darwin/compilers.yaml``.
-   The checked-in version worked for one developer but often needs to be changed for installed compiler
-   paths as well as `operating_system` to match your machine.
+.. code-block:: bash
+
+   $ cd /opt/local/bin
+   $ sudo ln -s mpicxx mpic++
+
+MacPorts will automatically update the ``mpicxx`` symlink in ``/opt/local/bin`` to point to the correct executable when you use the ``port select`` command to activate a partiular MPI package. 
+By making ``mpic++`` point to ``mpicxx``, this command will also automatically point to the correct executable if you change the global MPI package through MacPorts in the future.
+
+Python environment
+^^^^^^^^^^^^^^^^^^
+
+The Serac build scripts install the ``clingo`` package in your Python environment (and may even *uninstall* it if it finds it with a version it considers too old). 
+I don't like to let software packages change my Python environment, particularly by adding or removing packages with ``pip``, as this can lead to version dependency conflicts.  
+I create a Python virtual environment in the serac root directory with the built-in ``venv`` module. 
+
+.. (I use the name "venv" for the virtual environment; We might consider adding `venv/` to the list of git ignores in the root directory).
+
+.. code-block:: bash
+
+   $ python3 -m venv <virtual-env-name>
+
+Since spack will need ``clingo``, activate the python vitual environment and 
+install the package to meet this dependency:
+
+.. code-block:: bash
+
+   $ source <virtual-env-name>/bin/activate
+   $ pip install clingo
+
+The python virtual environment must be active when invoking the ``uberenv.py`` script below.
+
+Configuring spack
+^^^^^^^^^^^^^^^^^
+
+In ``compilers.yaml``:
+
+.. code-block:: yaml
+
+   - compiler:
```

Comment:
needs `compilers:` a line above `- compiler:`

Reply:
In the spirit of @white238's comment re: capturing existing institutional knowledge, this omission may be an artifact of using an older version of Spack configuration schema, which has changed at least once. 

IIRC, there are some automated tools to migrate config files using older schemas to the current schema. [`spack config update`](https://spack.readthedocs.io/en/latest/command_index.html#spack-config-update) looks like the relevant command (based on the context of https://github.com/spack/spack/issues/18031), but the documentation for that command is...sparse.

Reply:
Thanks. Fixed.

Reply:
might've just been a type-o, since the original `packages.yaml` file included the `compilers:` line.

i see what you mean about that command though. i tried using it for fun to see if it'd actually add the `compilers:` line itself. apparently these are the options (even though the documentation doesn't say): `compilers concretizer mirrors repos packages modules config upstreams bootstrap`

i ran
`spack config update configs`
...and got `==> No updates needed for "config" section.`